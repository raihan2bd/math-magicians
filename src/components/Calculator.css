.calc {
  width: 100%;
  max-width: 40rem;
  background: #fff;
}

.calc__input-group {
  width: 100%;
}

.calc__input {
  width: 100%;
  padding: 15px 8px;
  text-align: right;
  background: #858694;
  color: #fff;
  border: none;
  outline: none;
  font-size: 1rem;
}

.rows {
  list-style: none;
}

.row {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
}

.btn-default {
  padding: 15px;
  border: none;
  border-right: 1px solid #858694;
  border-bottom: 1px solid #858694;
  font-size: 0.9rem;
  font-weight: bold;
  background: #e0e0e0;
}

.btn-default:last-child {
  border-right: none;
}

.btn-orange {
  background: #f5913e;
}

.btn-zero {
  grid-column: 1/span 2;
}
