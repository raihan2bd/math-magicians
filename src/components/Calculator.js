import React from 'react';

import './Calculator.css';

export default class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="calc">
        <div className="calc__input-group">
          <input className="calc__input" type="text" value="0" readOnly />
        </div>
        <ul className="rows">
          <li className="row">
            <button className="btn-default" type="button">
              AC
            </button>
            <button className="btn-default" type="button">
              +/-
            </button>
            <button className="btn-default" type="button">
              %
            </button>
            <button className="btn-default btn-orange" type="button">
              &#247;
            </button>
          </li>
          <li className="row">
            <button className="btn-default" type="button">
              7
            </button>
            <button className="btn-default" type="button">
              8
            </button>
            <button className="btn-default" type="button">
              9
            </button>
            <button className="btn-default btn-orange" type="button">
              x
            </button>
          </li>
          <li className="row">
            <button className="btn-default" type="button">
              4
            </button>
            <button className="btn-default" type="button">
              5
            </button>
            <button className="btn-default" type="button">
              6
            </button>
            <button className="btn-default btn-orange" type="button">
              -
            </button>
          </li>
          <li className="row">
            <button className="btn-default" type="button">
              3
            </button>
            <button className="btn-default" type="button">
              2
            </button>
            <button className="btn-default" type="button">
              1
            </button>
            <button className="btn-default btn-orange" type="button">
              +
            </button>
          </li>
          <li className="row">
            <button className="btn-default btn-zero" type="button">
              0
            </button>
            <button className="btn-default" type="button">
              .
            </button>
            <button className="btn-default btn-orange" type="button">
              =
            </button>
          </li>
        </ul>
      </div>
    );
  }
}
