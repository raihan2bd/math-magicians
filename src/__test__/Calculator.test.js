import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import Calculator from '../components/Calculator';

describe('Testing Calculator component with snapshot', () => {
  test('testing Calculator component snapshot', () => {
    const calculator = renderer.create(<Calculator />).toJSON();

    expect(calculator).toMatchSnapshot();
  });

  test('test addition logic inside the Calculator component', () => {
    render(<Calculator />);

    const btn1 = screen.getByRole('button', { name: '1' });
    userEvent.click(btn1);
    const btnPlus = screen.getByRole('button', { name: '+' });
    userEvent.click(btnPlus);
    const btn2 = screen.getByRole('button', { name: '2' });
    userEvent.click(btn2);
    const btnEqual = screen.getByRole('button', { name: '=', exact: false });
    userEvent.click(btnEqual);

    const result = screen.getByTestId('total_value');

    expect(result.textContent).toBe('3');
  });

  test('test subtruction logic inside the Calculator component', () => {
    render(<Calculator />);

    const btn1 = screen.getByRole('button', { name: '1' });
    userEvent.click(btn1);
    userEvent.click(btn1);
    const operation = screen.getByRole('button', { name: '-' });
    userEvent.click(operation);
    const btn2 = screen.getByRole('button', { name: '6' });
    userEvent.click(btn2);
    const btnEqual = screen.getByRole('button', { name: '=', exact: false });
    userEvent.click(btnEqual);

    const result = screen.getByTestId('total_value');

    expect(result.textContent).toBe('5');
  });

  test('test multiplication logic inside the Calculator component', () => {
    render(<Calculator />);

    const btn1 = screen.getByRole('button', { name: '1' });
    userEvent.click(btn1);
    userEvent.click(btn1);
    const operation = screen.getByRole('button', { name: /x/i });
    userEvent.click(operation);
    const btn2 = screen.getByRole('button', { name: '2' });
    userEvent.click(btn2);
    const btnEqual = screen.getByRole('button', { name: '=', exact: false });
    userEvent.click(btnEqual);

    const result = screen.getByTestId('total_value');

    expect(result.textContent).toBe('22');
  });

  test('test Division logic inside the Calculator component', () => {
    render(<Calculator />);

    const btn1 = screen.getByRole('button', { name: '9' });
    userEvent.click(btn1);
    userEvent.click(btn1);
    const operation = screen.getByRole('button', { name: /รท/i });
    userEvent.click(operation);
    const btn2 = screen.getByRole('button', { name: '3' });
    userEvent.click(btn2);
    const btnEqual = screen.getByRole('button', { name: '=', exact: false });
    userEvent.click(btnEqual);

    const result = screen.getByTestId('total_value');

    expect(result.textContent).toBe('33');
  });
});
