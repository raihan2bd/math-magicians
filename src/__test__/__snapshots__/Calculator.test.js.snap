// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calculator component with snapshot testing Calculator component snapshot 1`] = `
<div
  className="card calc"
>
  <p
    className="calc__result"
    data-testid="total_value"
    type="text"
  >
    0
  </p>
  <ul
    className="rows"
  >
    <li
      className="row"
    >
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn-default btn-orange"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </li>
    <li
      className="row"
    >
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn-default btn-orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </li>
    <li
      className="row"
    >
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn-default btn-orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </li>
    <li
      className="row"
    >
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn-default btn-orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </li>
    <li
      className="row"
    >
      <button
        className="btn-default btn-zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn-default"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn-default btn-orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </li>
  </ul>
</div>
`;
